src/main/java/io/strimzi/kproxy/codec/OpaqueFrame.java

```diff
@@ -33,6 +33,12 @@ public abstract class OpaqueFrame implements Frame {
     public OpaqueFrame(ByteBuf buf, int length) {
         this.length = length;
         this.buf = buf.asReadOnly();
+        assert buf.readableBytes() == length;
+    }
+
+    @Override
+    public int estimateEncodedSize() {
+        return length + Integer.BYTES;
```

Comment:
Why do we need to add the extra Integer bytes? Worth a comment in the code, I think.

Reply:
Out encoding requires length prefix (according to the encode method), but yes I can add a comment to make it clear -> or we can use length prefix channel handlers too (exists in netty but that's another extra hop)

Reply:
Ah yes, I see. I'll take care of it and merge. Thanks!